### PR TITLE
Conversion to support both Python 2.7 and Python 3.5

### PR DESCRIPTION
**Pull request by danizen**
Hi,

I'm hoping this addresses #3 and #4.   I'm also hoping you are willing to act as a host to work on this software - I understand you've probably moved on, but if you can tell me how to run the various pieces, we can begin to get it back in working order.

If you look at my github profile, you'll be able to see my interest.   I've brought the OCCS/AB branch of the National Library of Medicine to Python, and as we upgrade our ILS, our cataloging/meta-data workflow depends on my getting this or some other well-supported MARC metasearch working with Python 3.

### CODE REVIEW

**Comment by danizen** [comment]
Well, it isn't working as well as I thought.   tox error.   Adding a travis-ci.org build is helping.  Give it a couple of days.

File: PyZ3950/CQLParser.py [hunk 1]
```diff
@@ -12,7 +12,7 @@
 from xml.dom.minidom import Node, parseString
 from PyZ3950.SRWDiagnostics import *
 # Don't use cStringIO as it borks Unicode (apparently)
-from StringIO import StringIO
+from io import StringIO
```
Comment (cclauss) [inline PyZ3950/CQLParser.py:15]: https://stackoverflow.com/questions/3410309/what-is-the-difference-between-stringio-and-io-stringio-in-python2-7

File: PyZ3950/CQLParser.py [hunk 2]
```diff
@@ -136,7 +136,7 @@ class ModifiableObject:
     modifiers = []
 
     def __getitem__(self, k):
-        if (type(k) == types.IntType):
+        if (type(k) == int):
```
Comment (cclauss) [inline PyZ3950/CQLParser.py:139]: PEP8: Use `isinstance()`

File: PyZ3950/c2query.py [hunk 3]
```diff
@@ -2,11 +2,10 @@
 #!/usr/local/bin/python2.3
 
 try:
-    from cStringIO import StringIO
+    from io import StringIO
```
Comment (cclauss) [inline PyZ3950/c2query.py:5]: See comment above.

This code says do X, if that does not work, do X again.

**Comment by danizen** [comment]
Timely, they just got our new Alma Z39.50 server running - I will get back to this.

File: PyZ3950/c2query.py [hunk 4]
```diff
@@ -1,10 +1,6 @@
-
 #!/usr/local/bin/python2.3
+from io import StringIO
```
Comment (cclauss) [inline PyZ3950/c2query.py:2]: In Python 2 `io.StringIO` != `StringIO.StringIO`
* https://stackoverflow.com/questions/3410309/what-is-the-difference-between-stringio-and-io-stringio-in-python2-7
```suggestion
try:
    from StringIO import StringIO  # Python 2
except ImportError:
    from io import StringIO  # Python 3
```

**Comment by mogul** [comment]
Hi there @cclauss and @danizen, what are the prospects for getting this PR merged any time soon?

**Comment by cclauss** [comment]
I am not a maintainer and have no rights to merge pull requests.

**Comment by mogul** [comment]
My apologies! 😊

@asl2, are you able to respond?

**Comment by asl2** [comment]
My apologies for not responding earlier.  I haven't done anything Z3950-related (or indeed library-related) in a while.
 
This is fine as far as it goes, but when I try to run under Python3 I get the following traceback:

Traceback (most recent call last):
  File "test1.py", line 33, in <module>
    main()
  File "test1.py", line 28, in main
    for r in get_results():
  File "test1.py", line 8, in get_results
    conn = zoom.Connection ('z3950.loc.gov', 7090)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/zoom.py", line 276, in __init__
    self.connect()
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/zoom.py", line 302, in connect
    optionslist = options, **initkw)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/z3950.py", line 511, in __init__
    ('initRequest', InitReq), 'initResponse')
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/z3950.py", line 529, in transact
    b = self.encode_ctx.encode (APDU, to_send)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 622, in encode
    spec.encode (self, data)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 1192, in encode
    ctyp.encode (ctx, val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 858, in encode
    self.typ.encode (ctx, val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 796, in encode
    self.encode_val (ctx, val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 1504, in encode_val
    typ.encode (ctx, v)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 858, in encode
    self.typ.encode (ctx, val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 796, in encode
    self.encode_val (ctx, val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 1089, in encode_val
    ctx.bytes_write (val)
  File "/usr/local/lib/python3.6/dist-packages/PyZ3950/asn1.py", line 633, in bytes_write
    raise EncodingError("Bad type to bytes_write")

Perhaps I've done something obvious wrong, but my thought was that PyZ3950 relied on Python 2's identity of string and bytes and would need somewhat more work for a port to Python 3.

I'll try to free up some time to look at this more closely.

**Comment by mogul** [comment]
I saw the commit fixing the tests. Thanks so much for taking care of this!

**Comment by cclauss** [comment]
Please see #6 for other Py3 changes needed.

**Pull request merged**